source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4460,42 +4464,51 @@ export class Source {
 }
 
 export class Media extends CardElement {
-    poster: string;
-    altText: string;
-    sources: Array<Source> = [];
     private _mediaType: string;
     private _mediaElement: HTMLMediaElement;
+	private _playButton: HTMLAnchorElement;
+	private _pauseButton: HTMLAnchorElement;
+	private _resetButton: HTMLAnchorElement;
+	private _poster: Image;
+
     static readonly acceptedMediaTypes = ["audio", "video"];
 
-    private processSources(): Array<Source> {
-        let typeFilter = (t): boolean => {
-            if (Media.acceptedMediaTypes.indexOf(t)) {
-                this._mediaType = t;
-                typeFilter = (tt) => tt === this._mediaType;
-                return true;
-            }
-            return false;
-        }
-        return this.sources.filter((s): boolean => {
-            if (s.mimeType) {
-                let mimeComponents = s.mimeType.split('/');
-                if (mimeComponents.length == 2) {
-                    return typeFilter(mimeComponents[0]);
-                }
-            }
-            return false;
-        });
+    private processSources(): Array<MediaSource> {
+		let goodSources: Array<MediaSource> = [];
+
+		for (let source of this.sources) {
+			let mimeComponents = source.mimeType.split('/');
+
+			if (mimeComponents.length == 2) {
+				if (!this._mediaType) {
+					let index = Media.acceptedMediaTypes.indexOf(mimeComponents[0]);
+
+					if (index >= 0) {
+						this._mediaType = Media.acceptedMediaTypes[index];
+					}
+				}
+
+				if (mimeComponents[0] == this._mediaType) {
+					goodSources.push(source);
+				}
+			} 
+		}
+
+		return goodSources;
     }
 
-    private makeButton(label: string): HTMLAnchorElement {
+    private static makeButton(label: string, buttonHint: string, handler: ()=>void): HTMLAnchorElement {
         let button = document.createElement("a");
-        button.className = "ac-media-button";
+        button.classList.add("ac-media-button", buttonHint);
+		button.title = buttonHint;
+		button.onclick = e => { handler(); e.preventDefault(); return false; };
```

Comment:
A couple things:

- Can you spread the statements over three lines below. In general that's better because it makes it possible to set a breakpoint on these statements.
- Event handlers are called onXxx in general in this codebase. Yours here would be onClick
- It would basically never happen because of the way you use makeButton, but ideall you would still check for null before invoking the handler.


Also, ActionButton do not call preventDefault in their handler, and that's a problem. Would you take this opportunity and fix that with this feature? Totally ok if you'd rather not.